File: mobile/src/main/kotlin/de/scout/fireplace/login/LoginClient.java
package de.scout.fireplace.login;

import de.scout.fireplace.models.Token;
import de.scout.fireplace.network.SchedulingStrategy;
import io.reactivex.Single;
import javax.inject.Inject;

class LoginClient {

  private static final String CLIENT_CREDENTIALS = "client_credentials";

  private final LoginService service;
  private final SchedulingStrategy strategy;

  @Inject
  LoginClient(LoginService service, SchedulingStrategy strategy) {
    this.service = service;
    this.strategy = strategy;
  }

  Single<Token> clientCredentials(String id, String secret) {
    return service.token(CLIENT_CREDENTIALS, id, secret)
        .compose(strategy.<Token>single());
  }
}

File: mobile/src/main/kotlin/de/scout/fireplace/login/LoginClient.kt
package de.scout.fireplace.login

import de.scout.fireplace.models.Token
import de.scout.fireplace.network.SchedulingStrategy
import io.reactivex.Single
import javax.inject.Inject

internal class LoginClient @Inject constructor(private val service: LoginService, private val strategy: SchedulingStrategy) {

  fun clientCredentials(id: String, secret: String): Single<Token> {
    return service.token(CLIENT_CREDENTIALS, id, secret).compose(strategy.single())
  }

  companion object {

    private val CLIENT_CREDENTIALS = "client_credentials"
  }
}
